### PR TITLE
API: Event system

### LINKED ISSUE #58: API: Create a functioning event system
Related to https://github.com/4fra/plutomierz/issues/38

### PR DESCRIPTION
**Pull request by Danrejk**
resolves #58 

This pull request includes several changes to enhance the event and vote handling system, improve the pluta value calculation, and refactor the WebSocket message handling. The most important changes are the addition of new database functions, updates to the pluta value calculation, and refactoring of WebSocket handlers.

### Enhancements to event and vote handling:

* [`api/src/db/handleEventsDb.ts`](diffhunk://#diff-135eb749503a95f0c7f39d92f9828eff5f6ad367918bc1e8e63e223a22c722e4R1-R61): Added functions `saveEventToDb` and `getLastEventsFromDb` to handle saving and fetching events from the database.
* [`api/src/db/handleVotesDb.ts`](diffhunk://#diff-76815c3b1f12ac266db67d7e8506aceb4cc21dad7088a9f5a532e9eb84352381R1-R40): Added function `getLastVotesFromDb` to fetch votes from the database and remove expired votes.
* [`api/src/db/initDb.ts`](diffhunk://#diff-cd9019c85bee945fc42cb195eff62614ef09293e847f71f35d71073789de06b6L15-R32): Updated the `events` table schema and added a new `votes` table schema.

### Improvements to pluta value calculation:

* [`api/src/plutaValue/getPlutaValue.ts`](diffhunk://#diff-f6e45e9b832feab2e4649404919c110d1ab6688b87e293c82fe807f90d497888R18-R19): Imported `accountVotes` and `accountEvents` utilities and updated the `getPlutaValue` function to account for events and votes in the pluta value calculation. [[1]](diffhunk://#diff-f6e45e9b832feab2e4649404919c110d1ab6688b87e293c82fe807f90d497888R18-R19) [[2]](diffhunk://#diff-f6e45e9b832feab2e4649404919c110d1ab6688b87e293c82fe807f90d497888R170-R174)
* [`api/src/plutaValue/utils/accountEvents.ts`](diffhunk://#diff-78c8f8a776936dd9c70f351a6205c64845083b84028dd49f484808bc43209443R1-R19): Added `accountEvents` function to adjust pluta value based on active events.
* [`api/src/plutaValue/utils/accountVotes.ts`](diffhunk://#diff-777e308e7fa943d3513924af26856e3b0b206a555b53928fd0169df2d9ed83d1R1-R18): Added `accountVotes` function to adjust pluta value based on active votes.

### Refactoring of WebSocket message handling:

* [`api/src/websocket/utils/handleChatMessage.ts`](diffhunk://#diff-24e1be4d804e210f898a49aabd7017598e225aa9ff71618512dfa47518855c32R1-R33): Added `handleChatMessage` function to handle chat messages, including validation and broadcasting to clients.
* [`api/src/websocket/utils/handleGetPlutaLog.ts`](diffhunk://#diff-9be1695f8e68f0825f180f5dcb1b0f58b03c90b0ab50f9c6a8d2fcb80b5d7dd7R1-R20): Added `handleGetPlutaLog` function to handle fetching and sending pluta logs to clients.
* [`api/src/websocket/websocket.ts`](diffhunk://#diff-1f16b70837f9bec466d186f82fff63a1544f48cc84d86be4c65a61977b9a69e8L4-R10): Refactored WebSocket connection handling to use `handleChatMessage` and `handleGetPlutaLog` functions. [[1]](diffhunk://#diff-1f16b70837f9bec466d186f82fff63a1544f48cc84d86be4c65a61977b9a69e8L4-R10) [[2]](diffhunk://#diff-1f16b70837f9bec466d186f82fff63a1544f48cc84d86be4c65a61977b9a69e8R30-R44) [[3]](diffhunk://#diff-1f16b70837f9bec466d186f82fff63a1544f48cc84d86be4c65a61977b9a69e8L58-R71)

These changes collectively improve the handling of events and votes, ensure accurate pluta value calculations, and streamline WebSocket message handling.

### CODE REVIEW

**Comment by Danrejk** [comment]
@4fra please check this to see if this fits your vision for the events

I made a draft pull request to let you comment on my progress.
**I haven't thoroughly checked everything,** but it seems to be working for me from what I've been testing

**Review by Beukot — APPROVED** [review]

File: api/src/db/initDb.ts [hunk 1]
```diff
@@ -12,16 +12,24 @@ const initDb = async () => {
 
         CREATE TABLE IF NOT EXISTS events (
             id INTEGER PRIMARY KEY AUTOINCREMENT,
-            user_id INTEGER,
+            user_id INTEGER NOT NULL,
```
Comment (4fra) [inline api/src/db/initDb.ts:15]: You need to edit existing table for this to work

Comment (Danrejk) [reply api/src/db/initDb.ts:15]: Thanks for reminding, i woudl forget

**Comment by Danrejk** [comment]
# TO DO BEFORE MERGE ON MASTER:
Manually edit the database to:
1. remove events table
2. let it create new one
3. manually add christmas event

**Review by 4fra — APPROVED** [review]

**Pull request merged**